dask/array/reshape.py

```diff
@@ -226,6 +231,33 @@ def reshape(x, shape, merge_chunks=True):
         x = x.rechunk({i: 1 for i in range(din - dout)})
 
     inchunks, outchunks = reshape_rechunk(x.shape, shape, x.chunks)
+    # Check output chunks are not too large
+    max_chunksize_in_bytes = reduce(mul, [max(i) for i in outchunks]) * x.dtype.itemsize
+    if limit is None and config.get("array.slicing.split-large-chunks") is not False:
+        limit = parse_bytes(config.get("array.chunk-size"))
+    if max_chunksize_in_bytes > limit:
+        # Leave chunk sizes unaltered where possible
+        matching_chunks = Counter(inchunks) & Counter(outchunks)
+        chunk_plan = []
+        for out in outchunks:
+            if out in matching_chunks:
+                chunk_plan.append(out)
+                # Remove match from list
+                if matching_chunks[out] == 1:
+                    matching_chunks.pop(out)
+                # Decrememt counter if multiple same-size matches exist
+                elif matching_chunks[out] > 1:
+                    matching_chunks[out] -= 1
```

Comment:
This could be simplified like:

```suggestion
            if matching_chunks[out] > 0:
                chunk_plan.append(out)
                matching_chunks[out] -= 1
```

because if the item isn't in the `Counter` it just gives 0

Reply:
Ah, that's much cleaner. Thank you for the suggestion!